server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexTemplateService.java

```diff
@@ -272,6 +280,21 @@ static ClusterState addIndexTemplateV2(final ClusterState currentState, final bo
             throw new IllegalArgumentException("index template [" + name + "] already exists");
         }
 
+        Map<String, List<String>> overlaps = findConflictingV1Templates(currentState, name, template.indexPatterns());
+        if (overlaps.size() > 0) {
```

Comment:
I think we should overlapping v2 templates with the same priority in the same way (but then throw an error) (in another change)

Reply:
Good idea, I'll add a TODO item to add this validation to the meta issue.